drake/systems/plants/allCollisionsmex.cpp

```diff
@@ -14,16 +14,17 @@ using namespace std;
 
 void mexFunction( int nlhs, mxArray *plhs[],int nrhs, const mxArray *prhs[] ) {
 
-  if (nrhs < 1) {
-    mexErrMsgIdAndTxt("Drake:collisionDetectmex:NotEnoughInputs","Usage collisionDetectmex(model_ptr)");
+  if (nrhs < 3) {
+    mexErrMsgIdAndTxt("Drake:allCollisions:NotEnoughInputs", "Usage allCollisionsmex(model_ptr, cache_ptr, active_collision_options)");
   }
 
-  // first get the model_ptr back from matlab
-  RigidBodyManipulator *model= (RigidBodyManipulator*) getDrakeMexPointer(prhs[0]);
+  int arg_num = 0;
+  RigidBodyManipulator *model = static_cast<RigidBodyManipulator*>(getDrakeMexPointer(prhs[arg_num++]));
+  KinematicsCache<double>* cache = static_cast<KinematicsCache<double>*>(getDrakeMexPointer(prhs[arg_num++]));
```

Comment:
and it feels (to me) like the opportunity to get these mixed up coming from the matlab side is even more severe than from the c++ side.  the matlab side allocates a new rbm somewhat behind the scenes (during a compile), so if someone did
```
model = RigidBodyManipulator(...)
kinsol = doKinematics();
model = compile(model);  % which could happen inside methods which are not labeled as obviously as compile
model.forwardKin(kinsol, ...);
````
then we would have a problem.  Which makes me realize that my proposal, if we go for it, should really be implemented (eventually) on the matlab side, too.  But I'd prefer to have the c++ side clean first.  What do people think?